src/crud__operation.ts

```diff
@@ -0,0 +1,207 @@
+  options: object,
+  updateDoc: object
+) => {
+  //=====create a filter
+  //const filter = { title: "Blacksmith Scene" };
+  //=====this option instructs the method to create a document if no documents match the filter
+  //const options = { upsert: true };
+  //=====create a document
+  //const updateDoc = {
+  //     $set: {
+  //       plot:
+  //         "Blacksmith Scene is a silent film directed by William K.L. Dickson",
+  //     },
+  //   };
+  const client = new MongoClient(lockalConnection, dbOptions);
+  try {
+    await client.connect();
+    const database = client.db("test");
+    const collection = database.collection("users");
+    return await collection.updateOne(filter, options, updateDoc);
+  } catch (error) {
+    console.log(error);
+  } finally {
+    await client.close();
+  }
+};
+
+//Udate many documents
+export const updateManyDoc = async (filter: object, updateDoc: object) => {
+  //=====create a filter
+  //const filter = { rated: "G" };
+  //=====increment every document matching the filter with 2 more comments
+  // const updateDoc = {
+  //     $inc: {
+  //       num_mflix_comments: 2,
+  //     },
+  //   };
+  const client = new MongoClient(lockalConnection, dbOptions);
+  try {
+    await client.connect();
+    const database = client.db("test");
+    const collection = database.collection("users");
+    return await collection.updateOne(filter, updateDoc);
+  } catch (error) {
+    console.log(error);
+  } finally {
+    await client.close();
+  }
+};
+
+//Replace one document
+export const replaceOneDoc = async (
+  query: object,
+  replacement: object,
+  options: object
+) => {
+  //=====Query
+  // const query = { title: "Blacksmith Scene" };
+  //=====create a new document that will be used to replace the existing document
+  //  const replacement = {
+  //     title: "Sandcastles in the Sand",
+  //     plot:
+  //       "Robin Sparkles mourns for a relationship with a mall rat at an idyllic beach.",
+  //   };
+  //=====Options
+  // const options = {
+  //     // create a document if no documents match the query
+  //     upsert: true,
+  //   };
+  const client = new MongoClient(lockalConnection, dbOptions);
+  try {
+    await client.connect();
+    const database = client.db("test");
+    const collection = database.collection("users");
+    return await collection.updateOne(query, replacement, options);
+  } catch (error) {
+    console.log(error);
+  } finally {
+    await client.close();
+  }
+};
+
+//Delet one document
+export const deleteOneDoc = async (query: object) => {
+  const client = new MongoClient(lockalConnection, dbOptions);
+  try {
+    await client.connect();
+    const database = client.db("test");
+    const collection = database.collection("users");
+    return await collection.deleteOne(query);
+  } catch (error) {
+    console.log(error);
+  } finally {
+    await client.close();
+  }
+};
+
+//Delet many documents
+export const deleteManyDoc = async (query: object) => {
```

Comment:
Better name will be `deleteManyDocuments`
Please replace this type `object` in something more related

Reply:
This file will by deleted. We not longer using this functions.